### ISSUE TITLE
5.1 channel audio stream is identified as 5 channel stream instead of 6 channel stream

### ISSUE THREAD

**Issue by Dnkhatri** [CastagnaIT/plugin.video.netflix#10]
*I'm submitting a ...*
  - [ x] bug report
  - [ ] feature request
  - [ ] support request

### Addon version used

14 beta 20

### Description
5.1 channel audio stream is identified as 5 channel stream instead of 6 channel stream so skins show it as 4.1 stream instead of the 5.1.

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

### Installation

* [ ] I'm using this Netflix Repo
* [ ] I'm using other Netflix Repo (Please specify which)
* [ ] I'm using a different source (Please tell which)

### Operating System

* [ ] Linux (x86/x64)
* [ ] OSX (x86/x64)
* [ ] Windows (x86/x64)
* [ ] Linux (ARM)
* [ ] Android

#### Additional informatin on the environment

[Descripe your environment a bit more detailed, are you using LibreElec f.e.]

## Debug log

[Please include a link to your debug log (use something like [http://sprunge.us/](http://sprunge.us/) or [https://pastebin.com/](https://pastebin.com/)) or similar, please do not paste]

## Other information

[e.g. detailed explanation, related issues, suggestions how to fix, links for us to have context, etc.]

### Screenshots

[Please add a screenshot if that helps understanding your problem]

[You can erase any parts of this template not applicable to your Issue.]

**Comment by CastagnaIT** [comment]
What are you talking about?
of kodi standard skins mediaflag? other external skins mediaflag? output problem to receiver? film/series that in official app info differs from kodi addon netflix?
What are you saying?...

**Comment by Essam315** [comment]
@CastagnaIT
I think he is talking about this problem.

https://github.com/asciidisco/plugin.video.netflix/pull/519#issuecomment-447636788

**Comment by CastagnaIT** [comment]
oh haha :D disgusting descriptions now i understand

**Comment by Essam315** [comment]
@CastagnaIT
The add-on file here can not be installed, need to be extracted and to zip the folder with WinRAR.

https://github.com/asciidisco/plugin.video.netflix/pull/519#issuecomment-464489174

see here

https://github.com/asciidisco/plugin.video.netflix/pull/519#issuecomment-464811198

https://github.com/asciidisco/plugin.video.netflix/pull/519#issuecomment-464821967

**Comment by CastagnaIT** [comment]
replaced thanks

**Comment by Essam315** [comment]
@CastagnaIT
You may need to tell the OP of this thread to replace the old "plugin.video.netflix-0.14.0.beta20.4K.zip" from the main page, with your master.

https://codeload.github.com/CastagnaIT/plugin.video.netflix/zip/master

https://forum.kodi.tv/showthread.php?tid=329767

**Comment by Essam315** [comment]
I have found a fix for this problem, in "converter.py" i replaced 

`def _convert_audio_downloadable(downloadable, adaptation_set, init_length,
                                channels_count):
    representation = ET.SubElement(
        parent=adaptation_set,
        tag='Representation',
        codecs='ec-3' if 'ddplus' in downloadable['content_profile'] else 'aac',
        bandwidth=str(downloadable['bitrate'] * 1024),
        mimeType='audio/mp4')
    ET.SubElement(
        parent=representation,
        tag='AudioChannelConfiguration',
        schemeIdUri='urn:mpeg:dash:23003:3:audio_channel_configuration:2011',
        value=str(channels_count))
    _add_base_url(representation, downloadable['urls'][0]['url'])
    _add_segment_base(representation, init_length)`

with

`def _convert_audio_downloadable(downloadable, adaptation_set, init_length,
                                channels_count):
    channels_count = {'1.0':'1', '2.0':'2', '5.1':'6', '7.1':'8'}				
    representation = ET.SubElement(
        parent=adaptation_set,
        tag='Representation',
        codecs='ec-3' if 'ddplus' in downloadable['content_profile'] else 'aac',
        bandwidth=str(downloadable['bitrate'] * 1024),
        mimeType='audio/mp4')
    ET.SubElement(
        parent=representation,
        tag='AudioChannelConfiguration',
        schemeIdUri='urn:mpeg:dash:23003:3:audio_channel_configuration:2011',
        value=channels_count[downloadable['channels']])
    _add_base_url(representation, downloadable['urls'][0]['url'])
    _add_segment_base(representation, init_length)`

You know that the channels number will be right when in "manifest.mpd"

`schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"`

and

`schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="6"`

**Comment by Essam315** [comment]
@CastagnaIT
I have found a fix for this problem, in "converter.py" i replaced 

`def _convert_audio_downloadable(downloadable, adaptation_set, init_length,
                                channels_count):
    representation = ET.SubElement(
        parent=adaptation_set,
        tag='Representation',
        codecs='ec-3' if 'ddplus' in downloadable['content_profile'] else 'aac',
        bandwidth=str(downloadable['bitrate'] * 1024),
        mimeType='audio/mp4')
    ET.SubElement(
        parent=representation,
        tag='AudioChannelConfiguration',
        schemeIdUri='urn:mpeg:dash:23003:3:audio_channel_configuration:2011',
        value=str(channels_count))
    _add_base_url(representation, downloadable['urls'][0]['url'])
    _add_segment_base(representation, init_length)`

with

`def _convert_audio_downloadable(downloadable, adaptation_set, init_length,
                                channels_count):
    channels_count = {'1.0':'1', '2.0':'2', '5.1':'6', '7.1':'8'}				
    representation = ET.SubElement(
        parent=adaptation_set,
        tag='Representation',
        codecs='ec-3' if 'ddplus' in downloadable['content_profile'] else 'aac',
        bandwidth=str(downloadable['bitrate'] * 1024),
        mimeType='audio/mp4')
    ET.SubElement(
        parent=representation,
        tag='AudioChannelConfiguration',
        schemeIdUri='urn:mpeg:dash:23003:3:audio_channel_configuration:2011',
        value=channels_count[downloadable['channels']])
    _add_base_url(representation, downloadable['urls'][0]['url'])
    _add_segment_base(representation, init_length)`

basically adding

`channels_count = {'1.0':'1', '2.0':'2', '5.1':'6', '7.1':'8'}	`

and replacing

`value=str(channels_count))`

with

`value=channels_count[downloadable['channels']])`
 

You know that the channels number will be right when in "manifest.mpd"

`schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"`

and

`schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="6"`

**Comment by CastagnaIT** [comment]
I've tried, but the descriptions are always the same

![image](https://user-images.githubusercontent.com/3257156/53040715-5fd99a80-3482-11e9-86b5-68c08f9c431f.png)

**Comment by CastagnaIT** [comment]
is to fix the flags of some skins?

**Comment by Essam315** [comment]
@CastagnaIT 
The problem is not in the description, the problem in the number of channels of the audio 5.1, without the fix you will get it as 5 channels, with the fix you will get the right number which is 6, see the photos here.

This is the right number of channels of 5.1 (6 channels).

https://user-images.githubusercontent.com/42877866/50053039-b2bb6c80-0136-11e9-8b0a-8f354ef30d72.png

and this is the wrong number of channels 5.1 (5 channels).

https://user-images.githubusercontent.com/42877866/50053099-9b30b380-0137-11e9-879b-edb27fb1cdd9.png

**Comment by CastagnaIT** [comment]
Thank you for the clarification! I never really noticed!
Thank you so much for your help
